assets/javascript/apps/pipeline/nodes/widgets.tsx

```diff
@@ -129,23 +132,95 @@ function ToggleWidget(props: WidgetParams) {
 
 function SelectWidget(props: WidgetParams) {
   const options = getSelectOptions(props.schema);
+  const selectedOption = options.find((option) => option.value.toString() === props.paramValue);
+  const [link, setLink] = useState<string | undefined>(selectedOption?.edit_url);
+
+  const onUpdate = (event: ChangeEvent<HTMLSelectElement>) => {
+    const selectedOption = options.find((option) => option.value.toString() === event.target.value);
+    setLink(selectedOption?.edit_url);
+    props.updateParamValue(event);
+  };
+  
+
   return <InputField label={props.label} help_text={props.helpText} inputError={props.inputError}>
-    <select
-      className="select select-bordered w-full"
-      name={props.name}
-      onChange={props.updateParamValue}
-      value={props.paramValue}
-      required={props.required}
-    >
+    <div className="flex flex-row gap-2">
+      <select
+        className="select select-bordered w-full"
+        name={props.name}
+        onChange={onUpdate}
+        value={props.paramValue}
+        required={props.required}
+      >
+        {options.map((option) => (
+          <option key={option.value} value={option.value}>
+            {option.label}
+          </option>
+        ))}
+      </select>
+      {link && (
+        <div className="tooltip" data-tip="Open in a new tab">
+          <a target="_blank" href={DOMPurify.sanitize(link)} className="align-bottom hover:cursor-pointer">
+            <i className="fa-solid fa-up-right-from-square fa-lg"></i>
+          </a>
+        </div>
+      )}
+    </div> 
+  </InputField>
+}
+
+
+function MultiSelectWidget(props: WidgetParams) {
+  const options = getSelectOptions(props.schema);
+  let selectedTools = Array.isArray(props.nodeParams["tools"]) ? props.nodeParams["tools"] : [];
```

Comment:
Wouldn't this be `props.paramValue`?

For this to be a generic widget you need to avoid using specific field names.

Reply:
https://github.com/dimagi/open-chat-studio/pull/950/commits/30463faa85276234463782358b0ca9eddddd88eb